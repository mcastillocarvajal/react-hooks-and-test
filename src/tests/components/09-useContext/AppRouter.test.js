import React from 'react';
import { mount } from 'enzyme';
import { AppRouter } from '../../../components/09-useContext/AppRouter';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('test for <AppRouter />', () => {

    const user = {
        id: 1,
        name: 'Moises'
    }

    const wrapper = mount( 
        <UserContext.Provider value={{
            user
        }}>
            <AppRouter /> );
        </UserContext.Provider>
    )

    test('should show comp correctly', () => {
        
        expect( wrapper ).toMatchSnapshot();    })
    
})
