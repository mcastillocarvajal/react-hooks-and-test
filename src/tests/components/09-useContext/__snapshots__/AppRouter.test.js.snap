// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for <AppRouter /> should show comp correctly 1`] = `
Array [
  <div>
    <nav
      className="navbar navbar-expand-sm navbar-dark bg-dark"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        useContext
      </a>
      <div
        className="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <a
            aria-current="page"
            className="nav-link active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home 
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </div>
      </div>
    </nav>
    <div
      className="container mt-3"
    >
      <div>
        <h1>
          HomeScreen
        </h1>
        <hr />
        <pre>
          {
   "id": 1,
   "name": "Moises"
}
        </pre>
      </div>
    </div>
  </div>,
  " );",
]
`;
